### DISCUSSION TITLE
Next broken after hard drive restore with error "normalScripts is not iterable"

### DISCUSSION THREAD

**Discussion by jeffwtm** [vercel/next.js#17131] [Help] [answered]
So this is a strange issue. I have a Next app that was working perfectly until a hard drive transfer/restore (on macOS), and now I consistently get a 500 Internal Server Error when trying to load it. I've tried reinstalling Next, all node modules, and even Docker, to no avail. Git confirms that there have been no changes since the last working commit.

The error I get is: TypeError: normalScripts is not iterable

It comes from the "getScripts" function of .next/server/pages/_document.js, specifically these lines:

`
    const {
      assetPrefix,
      files,
      buildManifest,
      isDevelopment
    } = this.context._documentProps;
    const {
      _devOnlyInvalidateCacheQueryString
    } = this.context;
    const normalScripts = files === null || files === void 0 ? void 0 : files.filter(file => file.endsWith('.js'));
    const lowPriorityScripts = (_buildManifest$lowPri = buildManifest.lowPriorityFiles) === null || _buildManifest$lowPri === void 0 ? void 0 : _buildManifest$lowPri.filter(file => file.endsWith('.js'));
    return [...normalScripts, ...lowPriorityScripts].map(file => {
`

The line that throws the error is the last one, because "normalScripts" is undefined. I modified the line two lines above to read `const normalScripts = files === null || files === void 0 ? [] : files.filter(file => file.endsWith('.js'));` which resolved the error. Adding a console.log of the _documentProps object confirmed that there is no "files" property present. My hack allows the page to load without an error, but no scripts or styles are loaded so it's non-functional.

Can anyone provide insight on where/how the "files" property is meant to be populated so I can continue investigating? I'm assuming this is somehow a permissions issue, or something along those lines. I can't think of anything else that makes sense. I'm using Next 9.5.0 if that helps (upgrading to 9.5.3 caused no change).

Thanks!

**Comment by jamesmosier** [comment, marked as answer]
Hi @jeffwtm. The version(s) of Next you tried no longer have `_documentProps`. See [this answer](https://github.com/vercel/next.js/discussions/16162#discussioncomment-53198) for more info.

You'd need to use Next 9.4.4 or earlier to get this to work. Otherwise the comment in that Discussion I linked has some other workarounds.

**Reply by jeffwtm** [reply]
Turns out I had an older version of Next still referenced in one of my libraries, and I guess it was pulling the _document.js file from there. Not entirely sure why the problem didn't show up until now - I suspect this points to an issue in my build process that I'll have to resolve. That said, knowing this helped me find and fix the problem, thanks very much!
